packages/next/types/index.d.ts

```diff
@@ -84,16 +97,33 @@ export type GetStaticPaths<
   fallback: boolean
 }>
 
-export type GetServerSideProps<
-  P extends { [key: string]: any } = { [key: string]: any },
-  Q extends ParsedUrlQuery = ParsedUrlQuery
-> = (context: {
+export type GetServerSidePropsContext<Q> = {
   req: IncomingMessage
   res: ServerResponse
   params?: Q
   query: ParsedUrlQuery
   preview?: boolean
   previewData?: any
-}) => Promise<{ props: P }>
+}
+
+export type GetServerSidePropsReturnType<P> = {
```

Comment:
Same as with `GetStaticPropsReturnType`, can we have a better name for this one?